swift_browser_ui_frontend/src/views/Upload.vue

```diff
@@ -626,56 +308,112 @@ export default {
     },
     // Make human readable translation functions available in instance
     // namespace
-    localHumanReadableSize: function ( size ) {
-      return getHumanReadableSize( size );
-    },
-    clearFiles() {
-      this.$store.commit("eraseDropFiles");
+    localHumanReadableSize: function (size) {
+      return getHumanReadableSize(size);
     },
     cancelUpload() {
       this.$store.commit("eraseDropFiles");
-      this.$router.go(-1);
+      this.$store.commit("toggleUploadModal", false);
+    },
+    dragHandler: function (e) {
+      e.preventDefault();
+      let dt = e.dataTransfer;
+      if (dt.types.indexOf("Files") >= 0) {
+        e.target.classList.add("over-dropArea");
+        e.stopPropagation();
+        dt.dropEffect = "copy";
+        dt.effectAllowed = "copy";
+      } else {
+        dt.dropEffect = "none";
+        dt.effectAllowed = "none";
+      }
+    },
+    dragLeaveHandler: function (e) {
+      e.target.classList.remove("over-dropArea");
+    },
+    navUpload: function (e) {
+      e.stopPropagation();
+      e.preventDefault();
+      if (e.dataTransfer && e.dataTransfer.items) {
+        for (let item of e.dataTransfer.items) {
+          this.$store.commit("appendFileTransfer", item);
+        }
+      } else if (e.dataTransfer && e.dataTransfer.files) {
+        for (let file of e.dataTransfer.files) {
+          this.$store.commit("appendFileTransfer", file);
+        }
+      }
+      e.target.classList.remove("over-dropArea");
     },
   },
 };
 </script>
 
-<style scoped>
-#uploadform {
-  width: 90%;
-  margin: auto;
+<style lang="scss" scoped>
+@import "@/css/prod.scss";
+
+.upload-card {
+  width: 64vw;
+  padding: 3rem;
+  left: 50%;
+  margin-top: 50%;
+  transform: translate(-50%, -50%);
```

Comment:
Translating this way seems to push the modal out-of-bounds on my machine, not sure if it's a low DPI monitor issue or what. Happens on both Firefox and Chrome, and with multiple different zoom levels.

![SD_Connect_OOB_Chrome](https://user-images.githubusercontent.com/24293337/181207631-9707b60e-f5ef-425c-89a9-ba1278963420.png)
![SD_Connect_OOB_Firefox](https://user-images.githubusercontent.com/24293337/181207666-5eaeec61-d13e-4c4d-b520-ac407d97ece5.png)

Reply:
I'll double check it's not due to having dev console open. I don't remeber it being so, but it seems to cause this for other modals